### DISCUSSION TITLE
Running JUnit 5 Suites with @SelectPackages

### DISCUSSION THREAD

**Discussion by drsgoodall** [eclipse-tycho/tycho#3664] [Q&A]
I'm having trouble with JUnit 5 test suites using @SelectPackages in particular. I can run them fine in Eclipse, but not with Tycho (4.0.6).

I have a test suite such as this,
```
@Suite
@SelectPackages({"com.example"})
public class SuiteTests {
}
```
However running in Tycho will result in 
```
WARNING: Error scanning files for URI bundleresource://11.fwk2141489356/com/example
java.nio.file.FileSystemNotFoundException: Provider "bundleresource" not installed
        at java.base/java.nio.file.Path.of(Path.java:212)
        at java.base/java.nio.file.Paths.get(Paths.java:98)
        at org.junit.platform.commons.util.CloseablePath.create(CloseablePath.java:58)
```
I've tried setting the `Eclipse-BundleShape` directive to dir, but this did not help.

I believe it might be the class loader - running in Eclipse  Thread.currentThread().getContextClassLoader() is `org.eclipse.pde.internal.junit.runtime.MultiBundleClassLoader` whereas in Tycho it is `class org.eclipse.tycho.surefire.osgibooter.CombinedClassLoader`

Does anyone have any suggestions on getting this working?

Thanks

**Comment by laeubi** [comment]
The best way always is to provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue or at least a small example project. What I rember is that suites have had problems in the past so there is a good chance that `@SelectPackages({"com.example"})` has similar issues.

**Comment by drsgoodall** [comment]
Hopefully this repository will illustrate the problem

https://github.com/drsgoodall/tycho-suite-selectpackages

Run with `mvn clean integration-test`

**Comment by Phillipus** [comment]
See https://github.com/eclipse-tycho/tycho/issues/3709
